net.sourceforge.pmd.eclipse.plugin/src/main/java/net/sourceforge/pmd/eclipse/runtime/properties/impl/ProjectPropertiesManagerImpl.java

```diff
@@ -76,378 +77,396 @@
+   */
+  @Override
+  public void storeProjectProperties(IProjectProperties projectProperties)
+      throws PropertiesException {
+    LOG.debug("Storing project properties for project " + projectProperties.getProject().getName());
+    try {
+      if (projectProperties.isPmdEnabled()) {
+        PMDNature.addPMDNature(projectProperties.getProject(), null);
+      } else {
+        PMDNature.removePMDNature(projectProperties.getProject(), null);
+      }
+
+      writeProjectProperties(projectProperties.getProject(), fillTransferObject(projectProperties));
+      projectsProperties.put(projectProperties.getProject(), projectProperties);
+
+    } catch (CoreException e) {
+      throw new PropertiesException("Core Exception when storing project properties for project "
+          + projectProperties.getProject().getName(), e);
     }
 
-    /**
-     * @see net.sourceforge.pmd.eclipse.runtime.properties.IProjectPropertiesManager#storeProjectProperties(net.sourceforge.pmd.eclipse.runtime.properties.IProjectProperties)
-     */
-    @Override
-    public void storeProjectProperties(IProjectProperties projectProperties) throws PropertiesException {
-        LOG.debug("Storing project properties for project " + projectProperties.getProject().getName());
-        try {
-            if (projectProperties.isPmdEnabled()) {
-                PMDNature.addPMDNature(projectProperties.getProject(), null);
-            } else {
-                PMDNature.removePMDNature(projectProperties.getProject(), null);
-            }
-
-            writeProjectProperties(projectProperties.getProject(), fillTransferObject(projectProperties));
-            projectsProperties.put(projectProperties.getProject(), projectProperties);
-
-        } catch (CoreException e) {
-            throw new PropertiesException("Core Exception when storing project properties for project "
-                    + projectProperties.getProject().getName(), e);
+  }
+
+  @Override
+  public void removeProjectProperties(IProject project) {
+    this.projectsProperties.remove(project);
+  }
+
+  /**
+   * Load the project rule set from the project ruleset
+   *
+   */
+  private void loadRuleSetFromProject(IProjectProperties projectProperties)
+      throws PropertiesException {
+    if (projectProperties.isRuleSetFileExist() && projectProperties.isNeedRebuild()) {
+      LOG.debug("Loading ruleset from project ruleset file: " + projectProperties.getRuleSetFile());
+      try {
+        final RuleSetFactory factory = new RuleSetFactory();
+
+        RuleSets allRulesets = new RuleSets();
+        for (final File ruleSetFile : projectProperties.getResolvedRuleSetFile()) {
+          RuleSet ruleSet = factory.createRuleSets(ruleSetFile.getPath()).getAllRuleSets()[0];
+          allRulesets.addRuleSet(ruleSet);
+          // This is the old rule stuff used to combine all the rulesets into one file.
```

Comment:
Delete the old implementation, we don't need it here. (It's still in the git history, if needed).